### PR TITLE
Improving Serverside Rendering Functionality & Updating Docker 

### PR DESCRIPTION
**Pull request by Schachte**
Helps improve SSR to fix some of the loading issues and improper bundle injection we were seeing on the serverside.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,7 +51,7 @@
     "polished": "^3.4.1",
     "react": "^16.9.0",
     "react-dom": "^16.9.0",
-    "react-loadable": "^5.5.0",
+    "react-loadable-ssr-addon": "https://github.com/schachte/react-loadable-ssr-addon#a88593a342fda85a79bf0dfc9f800fdff0693719",
```
Comment (djblue) [inline package.json]: I don't think revelio should bringing this in. I think ace should proxy.

Comment (Schachte) [reply package.json]: We use it directly in Revelio. 
https://github.com/connexta/revelio/pull/135/files#diff-1ac4767f66c2bb3895e13cd33163c951R10

File: package.json [hunk 2]
```diff
@@ -46,12 +46,12 @@
     "graphql-tools": "^4.0.5",
     "immutable": "^4.0.0-rc.12",
     "lodash.throttle": "^4.1.1",
+    "memfs": "^3.0.1",
```
Comment (Schachte) [inline package.json]: remove this

File: src/main/graphql-server/helpers/renderer.js [hunk 3]
```diff
@@ -26,60 +21,94 @@ module.exports = async (req, res, next) => {
...
+        <script
+          dangerouslySetInnerHTML={{
+            __html: `window.__APOLLO_STATE__ = ${JSON.stringify(state)}`,
+          }}
+        />
+        <style id="css-server-side" dangerouslySetInnerHTML={{ __html: css }} />
+      </head>
+      <body style={{ margin: 0 }}>
+        <div id="root" dangerouslySetInnerHTML={{ __html: content }} />
+        {/* estlint-ignore-next-line */}
```
Comment (Schachte) [inline src/main/graphql-server/helpers/renderer.js]: remove this

File: src/main/graphql-server/helpers/renderer.js [hunk 4]
```diff
@@ -26,60 +21,94 @@ module.exports = async (req, res, next) => {
...
+          </StaticRouter>
+        </ApolloProvider>
+      </Loadable.Capture>
     )
-  }
-
-  try {
     await getDataFromTree(App)
-    await sleep(300)
+    await sleep(500)
```
Comment (Schachte) [inline src/main/graphql-server/helpers/renderer.js]: Add todo to look into this race condition stuff

File: src/main/graphql-server/server.js [hunk 5]
```diff
@@ -5,10 +5,24 @@ const router = require('./middleware')
...
+// Memoize the manifest to reduce disk reads
+let cachedBundleManifest = null
+app.use('/', (req, res, next) => {
+  if (cachedBundleManifest == null) {
+    cachedBundleManifest = fs.readFileSync(
+      '../webapp/react-loadable.json',
+      'utf8'
+    )
+  }
+  req.clientBundles = JSON.parse(cachedBundleManifest)
```
Comment (Schachte) [inline src/main/graphql-server/server.js]: cache the parsed version

File: src/main/webapp/user.js [hunk 6]
```diff
@@ -47,7 +47,7 @@ const User = props => {
   }
 
   const signOut = () => {
-    window.location.href = '../../logout?service=' + window.location.href
+    // window.location.href = '../../logout?service=' + window.location.href
```
Comment (djblue) [inline src/main/webapp/user.js]: Undo

File: src/main/graphql-server/server.js [hunk 7]
```diff
@@ -5,10 +5,23 @@ const router = require('./middleware')
...
+const fs = require('fs')
 
 app.use(compression())
+
+// Memoize the manifest to reduce disk reads
+let cachedBundleManifest = null
+app.use('/', (req, res, next) => {
+  if (cachedBundleManifest == null) {
+    cachedBundleManifest = JSON.parse(
+      fs.readFileSync('../webapp/react-loadable.json', 'utf8')
```
Comment (djblue) [inline src/main/graphql-server/server.js]: Do we need a __dirname here?

**Pull request merged**